<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.11"/>
<title>GGTK: The GO Graph Took Kit for working with the Gene Ontology: NCList Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { init_search(); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">GGTK: The GO Graph Took Kit for working with the Gene Ontology
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.11 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classNCList.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#nested-classes">Classes</a> &#124;
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="classNCList-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">NCList Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>A container class for quickly finding intersections of genomic intervals.  
 <a href="classNCList.html#details">More...</a></p>

<p><code>#include &lt;<a class="el" href="NCList_8hpp_source.html">ggtk/NCList.hpp</a>&gt;</code></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Classes</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">class &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList_1_1NCNode.html">NCNode</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:ab6c2b7decd70fcde7b54db4e79b64b85"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#ab6c2b7decd70fcde7b54db4e79b64b85">NCList</a> (std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt; &amp;regions)</td></tr>
<tr class="separator:ab6c2b7decd70fcde7b54db4e79b64b85"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a98e796a8e78efec9d019fb4303d4c8f3"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a98e796a8e78efec9d019fb4303d4c8f3">getOverlapIndex</a> (const <a class="el" href="classGenomicRegion.html">GenomicRegion</a> r)</td></tr>
<tr class="separator:a98e796a8e78efec9d019fb4303d4c8f3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3199031fe2904e2cac57a682e07573c5"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a3199031fe2904e2cac57a682e07573c5">getOverlapIndex</a> (const std::pair&lt; size_t, size_t &gt; p)</td></tr>
<tr class="separator:a3199031fe2904e2cac57a682e07573c5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a01ce26e143d1edf32caf50d472c1cce3"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a01ce26e143d1edf32caf50d472c1cce3">getOverlapIndex</a> (const size_t start, const size_t end)</td></tr>
<tr class="separator:a01ce26e143d1edf32caf50d472c1cce3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a226ca89f140feac5088af4be366bcd58"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a226ca89f140feac5088af4be366bcd58">getOverlapIndex</a> (const size_t point)</td></tr>
<tr class="separator:a226ca89f140feac5088af4be366bcd58"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af67a0c748ed4e7d095bb6092c1f291fb"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#af67a0c748ed4e7d095bb6092c1f291fb">getFeatures</a> ()</td></tr>
<tr class="separator:af67a0c748ed4e7d095bb6092c1f291fb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acaec79f5de348b5645a856fcde6d58cf"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#acaec79f5de348b5645a856fcde6d58cf">getFeaturesInRange</a> (const <a class="el" href="classGenomicRegion.html">GenomicRegion</a> r)</td></tr>
<tr class="separator:acaec79f5de348b5645a856fcde6d58cf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aab695768230d12b179699ca8f19f5d4c"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#aab695768230d12b179699ca8f19f5d4c">getFeaturesInRange</a> (const std::pair&lt; size_t, size_t &gt; p)</td></tr>
<tr class="separator:aab695768230d12b179699ca8f19f5d4c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a02b767b23f0007af67174cb2c2e6d92b"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a02b767b23f0007af67174cb2c2e6d92b">getFeaturesInRange</a> (const size_t start, const size_t end)</td></tr>
<tr class="separator:a02b767b23f0007af67174cb2c2e6d92b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6de14b47eb6b5c7291f5469609d500f6"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a6de14b47eb6b5c7291f5469609d500f6">getFeaturesAt</a> (const size_t index)</td></tr>
<tr class="separator:a6de14b47eb6b5c7291f5469609d500f6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4bb83c717bfa1b169f48809a4ed32f2e"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a4bb83c717bfa1b169f48809a4ed32f2e">size</a> ()</td></tr>
<tr class="separator:a4bb83c717bfa1b169f48809a4ed32f2e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a88ba10ba7017497f6912e8f3570239e9"><td class="memItemLeft" align="right" valign="top">size_t&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a88ba10ba7017497f6912e8f3570239e9">numFeatures</a> ()</td></tr>
<tr class="separator:a88ba10ba7017497f6912e8f3570239e9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4ea0b59d58fcd385adb2fa4e92aae35b"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classNCList_1_1NCNode.html">NCNode</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a4ea0b59d58fcd385adb2fa4e92aae35b">getNodes</a> ()</td></tr>
<tr class="separator:a4ea0b59d58fcd385adb2fa4e92aae35b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a002aee45aea558b91957c4f282b83565"><td class="memItemLeft" align="right" valign="top"><a class="el" href="classGenomicRegion.html">GenomicRegion</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a002aee45aea558b91957c4f282b83565">getRegionAt</a> (size_t i)</td></tr>
<tr class="separator:a002aee45aea558b91957c4f282b83565"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:a2999e47110bf2e20bb9fd8e8689715e3"><td class="memItemLeft" align="right" valign="top">std::vector&lt; <a class="el" href="classNCList_1_1NCNode.html">NCNode</a> &gt;&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classNCList.html#a2999e47110bf2e20bb9fd8e8689715e3">_items</a></td></tr>
<tr class="separator:a2999e47110bf2e20bb9fd8e8689715e3"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>A container class for quickly finding intersections of genomic intervals. </p>
<p>This class prepresents an implementation of Nested Containment Lists by Alekseyenko and Lee.</p>
<p>Alekseyenko AV and Lee CJ. Nested Containment List (<a class="el" href="classNCList.html" title="A container class for quickly finding intersections of genomic intervals. ">NCList</a>): a new algorithm for accelerating interval query of genome alignment and interval databases. Bioinformatics. 2007 Jun 1;23(11):1386-93. Epub 2007 Jan 18. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="ab6c2b7decd70fcde7b54db4e79b64b85"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">NCList::NCList </td>
          <td>(</td>
          <td class="paramtype">std::vector&lt; <a class="el" href="classGenomicRegion.html">GenomicRegion</a> &gt; &amp;&#160;</td>
          <td class="paramname"><em>regions</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Constructor for <a class="el" href="classNCList.html" title="A container class for quickly finding intersections of genomic intervals. ">NCList</a> that performs the work of sorting and inserting intevals into the NCNodes. </p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="af67a0c748ed4e7d095bb6092c1f291fb"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classGenomicRegion.html">GenomicRegion</a>&gt; NCList::getFeatures </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>A method to return all genes as a vector </p>

</div>
</div>
<a class="anchor" id="a6de14b47eb6b5c7291f5469609d500f6"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classGenomicRegion.html">GenomicRegion</a>&gt; NCList::getFeaturesAt </td>
          <td>(</td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>index</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method to return all genes at a particular index in the list </p>

</div>
</div>
<a class="anchor" id="acaec79f5de348b5645a856fcde6d58cf"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classGenomicRegion.html">GenomicRegion</a>&gt; NCList::getFeaturesInRange </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classGenomicRegion.html">GenomicRegion</a>&#160;</td>
          <td class="paramname"><em>r</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method to return all genes in a given range </p>

</div>
</div>
<a class="anchor" id="aab695768230d12b179699ca8f19f5d4c"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classGenomicRegion.html">GenomicRegion</a>&gt; NCList::getFeaturesInRange </td>
          <td>(</td>
          <td class="paramtype">const std::pair&lt; size_t, size_t &gt;&#160;</td>
          <td class="paramname"><em>p</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method to return all genes between a start end pair </p>

</div>
</div>
<a class="anchor" id="a02b767b23f0007af67174cb2c2e6d92b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classGenomicRegion.html">GenomicRegion</a>&gt; NCList::getFeaturesInRange </td>
          <td>(</td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>start</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>end</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Method to return all genes between a start end interval </p>

</div>
</div>
<a class="anchor" id="a4ea0b59d58fcd385adb2fa4e92aae35b"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classNCList_1_1NCNode.html">NCNode</a>&gt; NCList::getNodes </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>return the vector of NCNodes </p>

</div>
</div>
<a class="anchor" id="a98e796a8e78efec9d019fb4303d4c8f3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::getOverlapIndex </td>
          <td>(</td>
          <td class="paramtype">const <a class="el" href="classGenomicRegion.html">GenomicRegion</a>&#160;</td>
          <td class="paramname"><em>r</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Get the list index of the overlapped region </p>

</div>
</div>
<a class="anchor" id="a3199031fe2904e2cac57a682e07573c5"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::getOverlapIndex </td>
          <td>(</td>
          <td class="paramtype">const std::pair&lt; size_t, size_t &gt;&#160;</td>
          <td class="paramname"><em>p</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Get the list index of the overlapped start end pair </p>

</div>
</div>
<a class="anchor" id="a01ce26e143d1edf32caf50d472c1cce3"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::getOverlapIndex </td>
          <td>(</td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>start</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>end</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Get the list index of the overlapped start end interval </p>

</div>
</div>
<a class="anchor" id="a226ca89f140feac5088af4be366bcd58"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::getOverlapIndex </td>
          <td>(</td>
          <td class="paramtype">const size_t&#160;</td>
          <td class="paramname"><em>point</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Get the list index of the overlapped coordinate </p>

</div>
</div>
<a class="anchor" id="a002aee45aea558b91957c4f282b83565"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname"><a class="el" href="classGenomicRegion.html">GenomicRegion</a> NCList::getRegionAt </td>
          <td>(</td>
          <td class="paramtype">size_t&#160;</td>
          <td class="paramname"><em>i</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>returns the top level region at an index i </p>

</div>
</div>
<a class="anchor" id="a88ba10ba7017497f6912e8f3570239e9"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::numFeatures </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>numFeatures returns number of all genes/features </p>

</div>
</div>
<a class="anchor" id="a4bb83c717bfa1b169f48809a4ed32f2e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">size_t NCList::size </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">inline</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>size returns number of top level nodes </p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a2999e47110bf2e20bb9fd8e8689715e3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::vector&lt;<a class="el" href="classNCList_1_1NCNode.html">NCNode</a>&gt; NCList::_items</td>
        </tr>
      </table>
</div><div class="memdoc">
<p>List of NCNodes as a vector </p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>ggtk/<a class="el" href="NCList_8hpp_source.html">NCList.hpp</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="classNCList.html">NCList</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.11 </li>
  </ul>
</div>
</body>
</html>
